src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs

```diff
@@ -47,12 +47,12 @@ protected override bool ReceiveRecover(object message)
             if (message is EntityStarted)
             {
                 var started = (EntityStarted)message;
-                State = new ShardState(State.Entries.Add(started.EntityId));
+                State = new ShardState(State.Entries?.Add(started.EntityId));
```

Comment:
This is sign of another issue, which lies [here](https://github.com/akkadotnet/akka.net/blob/dev/src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs#L57). I suspect, that the reason behind that may be that `ShardState` is struct. By default when no snapshot has been found, it's `SnapshotOffer.Snapshot` field should contain null value. But could it be possible that in case of structs, that field could be `default(ShardState)` instead? In that case all null checks would be omitted while still passing an invalid state (with uninitialized entries) through.